library/src/image.cxx

```diff
@@ -492,14 +500,25 @@ const image& image::save(const std::string& path, SaveFormat format) const
       break;
   }
 
-  writer->SetFileName(path.c_str());
-  writer->SetInputData(this->Internals->Image);
-  writer->Write();
+  try
+  {
+    // Ensure the directories exists
+    fs::create_directories(filePath.parent_path());
+
+    writer->SetFileName(filePath.string().c_str());
+    writer->SetInputData(this->Internals->Image);
+    writer->Write();
 
-  if (writer->GetErrorCode() != 0)
+    if (writer->GetErrorCode() != 0)
+    {
+      throw write_exception("Cannot write " + filePath.string());
```

Comment:
```suggestion
      throw write_exception("Cannot write image " + filePath.string());
```
or
```suggestion
      throw write_exception("Cannot write file " + filePath.string());
```
not sure which one, there's both a `throw read_exception("Cannot open file " + filePath.string());` and a `throw read_exception("Cannot read image " + filePath.string());` on the reading side